当运行中的线程抛出异常的时候,会释放掉持有的所有监视器的所有权.  
```java
package ocn.axy.concurrent;  
import java.util.logging.Level;  
import java.util.logging.Logger;  
public class ThreadApp {  
    private final static Logger lg = Logger.getLogger(ThreadApp.class.getPackage().getName());  
    // 特性4：当前线程持有监视器的所有权，若当前线程抛出异常，会释放所有持有的监视器！  
    public static void main(String[] args) {  
        try {  
            Plan plan = new Plan();  
            plan.start();// 该线程未抛出异常之前，会一直持有监视器的所有权。  
            Thread.sleep(500);  
            plan.m2();// plan线程抛出异常，处于阻塞状态的main线程获取了监视器的所有权。  
        } catch (InterruptedException ex) {  
            lg.log(Level.WARNING, ex.toString(), ex);  
        }  
    }  
    private static class Plan extends Thread {  
        @Override  
        public void run() {  
            try {  
                m1();  
            } catch (Exception ex) {  
                lg.log(Level.WARNING, ex.toString(), ex);  
            }  
          }  
          public synchronized void m1() throws Exception {  
              long base = System.currentTimeMillis();  
              long now = 0;  
              for (;;) {  
                  now = System.currentTimeMillis();  
                  if (now - base == 3_000)  
                      break;  
              }  
              throw new Exception();  
          }  
          public synchronized void m2() {  
              lg.info("method m2");  
          }  
      }  
  }  
```  
```java
星期四 十一月 21 16:44:27 CST 2019 ocn.axy.concurrent.ThreadApp$Plan m2  
信息: method m2  
星期四 十一月 21 16:44:27 CST 2019 ocn.axy.concurrent.ThreadApp$Plan run
警告: java.lang.Exception  
java.lang.Exception  
    at ocn.axy.concurrent.ThreadApp$Plan.m1(ThreadApp.java:34)  
    at ocn.axy.concurrent.ThreadApp$Plan.run(ThreadApp.java:21)  
```  
